### ISSUE TITLE
语音衔接速度过快

### ISSUE THREAD

**Issue by JY666666** [Soul-AILab/SoulX-Podcast#16]
现在生成的对话语音，S1 说完，S2 立刻就接上了，给人听的感觉很赶，有办法加停顿吗？

**Comment by grootgordon** [comment]
> [#22](https://github.com/Soul-AILab/SoulX-Podcast/pull/22) feat: 增强对话合成功能，支持内联停顿标记并优化停顿时间控制 <|pause:MS|> feat: 对话人之间，增加停顿时间控制；待生成的文本支持换行
> 
> <img alt="PixPin_2025-11-05_23-40-28" width="1945" height="1368" src="https://private-user-images.githubusercontent.com/49787303/510226106-f4215a54-8dc2-47bc-af58-6cdc1343b74d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc2MjU1OTEsIm5iZiI6MTc2NzYyNTI5MSwicGF0aCI6Ii80OTc4NzMwMy81MTAyMjYxMDYtZjQyMTVhNTQtOGRjMi00N2JjLWFmNTgtNmNkYzEzNDNiNzRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA1VDE1MDEzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQwYWZlZTFhMGI2YTk3YjEzYWYyMzA0ZDAzNmIzZDY4ZjE1MjQxZWI4YjZiYWVkMDc1NjNjYjZhOTc5YjdlOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8ReFQ5HXSjm_g4Q9T2YkWVMUGHcFQLaS6SnByosaA0c">

兄弟，你这个pr尚未合并，我改完了代码跑出来mp3,听到pause才发现服务不支持:(

**Comment by thesnail** [comment]
> > [#22](https://github.com/Soul-AILab/SoulX-Podcast/pull/22) feat: 增强对话合成功能，支持内联停顿标记并优化停顿时间控制 <|pause:MS|> feat: 对话人之间，增加停顿时间控制；待生成的文本支持换行
> > <img alt="PixPin_2025-11-05_23-40-28" width="1945" height="1368" src="https://private-user-images.githubusercontent.com/49787303/510226106-f4215a54-8dc2-47bc-af58-6cdc1343b74d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc2MjU1OTEsIm5iZiI6MTc2NzYyNTI5MSwicGF0aCI6Ii80OTc4NzMwMy81MTAyMjYxMDYtZjQyMTVhNTQtOGRjMi00N2JjLWFmNTgtNmNkYzEzNDNiNzRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA1VDE1MDEzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQwYWZlZTFhMGI2YTk3YjEzYWYyMzA0ZDAzNmIzZDY4ZjE1MjQxZWI4YjZiYWVkMDc1NjNjYjZhOTc5YjdlOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8ReFQ5HXSjm_g4Q9T2YkWVMUGHcFQLaS6SnByosaA0c">
> 
> 兄弟，你这个pr尚未合并，我改完了代码跑出来mp3,听到pause才发现服务不支持:(

兄弟,现在可以了吗

**Comment by thesnail** [comment]
> import os import json import torch import argparse import random
> 
> import s3tokenizer import soundfile as sf
> 
> from soulxpodcast.config import SamplingParams from soulxpodcast.utils.parser import podcast_format_parser from soulxpodcast.utils.infer_utils import initiate_model, process_single_input
> 
> def run_inference( inputs: dict, model_path: str, output_path: str, llm_engine: str = "hf", fp16_flow: bool = False, seed: int = 1988, ):
> 
> ```
> model, dataset = initiate_model(seed, model_path, llm_engine, fp16_flow)
> 
> data = process_single_input(
>     dataset,
>     inputs['text'],
>     inputs['prompt_wav'],
>     inputs['prompt_text'],
>     inputs['use_dialect_prompt'],
>     inputs['dialect_prompt_text'],
> )
> 
> print("[INFO] Start inference...")
> results_dict = model.forward_longform(**data)
> 
> target_audio = None
> sample_rate = 24000  # 音频采样率
> for idx, wav in enumerate(results_dict["generated_wavs"]):
>     if target_audio is None:
>         target_audio = wav
>     else:
>         # 在对话之间添加随机停顿（0.3-1.0秒）
>         pause_duration = random.uniform(0.8, 2.0)
>         pause_samples = int(pause_duration * sample_rate)
>         silence = torch.zeros(1, pause_samples).to(wav.device)
>         
>         # 拼接：已有音频 + 停顿 + 新音频
>         target_audio = torch.cat([target_audio, silence, wav], dim=1)
>         print(f"[INFO] Added {pause_duration:.2f}s pause before dialogue {idx+1}")
> 
> os.makedirs(os.path.dirname(output_path), exist_ok=True)
> sf.write(output_path, target_audio.cpu().squeeze(0).numpy(), 24000)
> print(f"[INFO] Saved synthesized audio to: {output_path}")
> ```
> 
> if **name** == "**main**": parser = argparse.ArgumentParser() parser = argparse.ArgumentParser() parser.add_argument("--json_path", required=True, help="Path to the input JSON file") parser.add_argument("--model_path", required=True, help="Path to the model file") parser.add_argument("--output_path", default="outputs/result.wav", help="Path to the output audio file") parser.add_argument("--llm_engine", default="hf", choices=["hf", "vllm"], help="Inference engine to use") parser.add_argument("--fp16_flow", action="store_true", help="Enable FP16 flow") parser.add_argument("--seed", type=int, default=1988, help="Random seed") args = parser.parse_args()
> 
> ```
> with open(args.json_path, "r") as f:
>     data = json.load(f)
> inputs = podcast_format_parser(data)
> run_inference(
>     inputs=inputs,
>     model_path=args.model_path,
>     output_path=args.output_path,
>     llm_engine=args.llm_engine,
>     fp16_flow=args.fp16_flow,
>     seed=args.seed,
> )//拿这个替换你的SoulX-Podcast/cli/podcast.py脚本
> ```

好像没有生效
